apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1/UrlSigner.cs

```diff
@@ -432,4 +454,100 @@ private static JsonCredentialParameters ParseCredentialParameters(Stream credent
             return credentialParameters;
         }
     }
+
+    public sealed class SignedUrlUploader : Apis.Upload.TmpResumableUpload
+    {
+        private SignedUrlUploader(Uri sessionUri, Stream contentStream, ConfigurableHttpClient httpClient)
+        {
+            UploadUri = sessionUri;
+            ContentStream = contentStream;
+            HttpClient = httpClient ?? new ConfigurableHttpClient(new ConfigurableMessageHandler(new HttpClientHandler()));
+        }
+
+        protected override Task<GoogleApiException> ExceptionForResponseAsync(HttpResponseMessage response)
+        {
+            return ExceptionForResponseInternalAsync(response);
+        }
+
+        private static Task<GoogleApiException> ExceptionForResponseInternalAsync(HttpResponseMessage response)
+        {
+            return MediaApiErrorHandling.ExceptionForResponseAsync(NewtonsoftJsonSerializer.Instance, "storage", response);
+        }
+
+        protected override Task<Uri> InitializeUpload(CancellationToken cancellationToken)
+        {
+            return Task.FromResult(UploadUri);
+        }
+
+        public static SignedUrlUploader CreateFromSessionUri(
+            Uri sessionUri,
+            Stream contentStream,
+            ConfigurableHttpClient httpClient = null)
+        {
+            return new SignedUrlUploader(sessionUri, contentStream, httpClient);
+        }
+
+        public static SignedUrlUploader CreateFromSignedUrl(
+            string signedUrl,
+            Stream contentStream,
+            ConfigurableHttpClient httpClient = null,
+            Action<HttpRequestMessage> modifySessionUriRequest = null)
+        {
+            return GetResult(token => CreateFromSignedUrlAsync(signedUrl, contentStream, httpClient, modifySessionUriRequest, token));
+        }
+
+        public static async Task<SignedUrlUploader> CreateFromSignedUrlAsync(
+            string signedUrl,
+            Stream contentStream,
+            ConfigurableHttpClient httpClient = null,
```

Comment:
Ah, yes... if they don't have any credentials, they can't create a StorageService. Hmm. Yes, that's reasonable... and presumably most of the time we expect users to just leave out this parameter anyway? Perhaps we should have an overload for that common case rather than relying on optional parameters, to avoid anxiety from users who worry about what `ConfigurableHttpClient` is...